src/api.v2/routes/kubernetes.js

```diff
@@ -0,0 +1,33 @@
+const AWSXRay = require('aws-xray-sdk');
+const k8s = require('@kubernetes/client-node');
+const getLogger = require('../../utils/getLogger');
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+
+const logger = getLogger();
+
+module.exports = {
+  'kubernetes#event': async (req, res, next) => {
+    logger.log('received kubernetes event');
+    try {
+      const {
+        reason, message, type, involvedObject: { name, namespace },
+      } = req.body;
+      logger.log(`[${reason}] received ${type} kubernetes event: ${message} ${name} in ${namespace}`);
+
+      // remove only pods in your namespace and due to backoff errors
+      if ((namespace.match('^pipeline-.*') || namespace.match('^worker-.*'))
+       && reason === 'BackOff' && type !== 'Normal' && message.includes('restarting')) {
+        const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
+        logger.log(`removing pod ${name} in ${namespace}`);
+        await k8sApi.deleteNamespacedPod(name, namespace);
+      }
+      res.status(200).send('ok');
```

Comment:
We have the `OK()` helper in the API (`src/utis/responses/OK`). Can you use that instead of hard-coding this?

Reply:
I prefer not to, they are not equivalent and the `OK()` should still be included so it would be `res.json(OK())` because we are in `routes` instead of `route-services` where the naked `OK()` is used. I had a lot of issues with the `event-exporter` due to the responses we were sending to it, so I'd rather leave it as it is considering it's an isolated and exceptional endpoint.

Reply:
I modified it in the end, we'll need to verify it works correctly with the `event-exporter`